source/common/redis/proxy_filter.cc

```diff
@@ -19,9 +19,15 @@ ProxyFilterConfig::ProxyFilterConfig(const Json::Object& config, Upstream::Clust
       cluster_name_(config.getString("cluster_name")),
       stat_prefix_(fmt::format("redis.{}.", config.getString("stat_prefix"))),
       stats_(generateStats(stat_prefix_, scope)) {
-  if (!cm.get(cluster_name_)) {
-    throw EnvoyException(
-        fmt::format("redis filter config: unknown cluster name '{}'", cluster_name_));
+  Upstream::ThreadLocalCluster* cluster = cm.get(cluster_name_);
+  if (cluster == nullptr) {
+    throw EnvoyException(fmt::format("redis filter config: unknown cluster '{}'", cluster_name_));
+  }
+
+  if (cluster->info()->addedViaApi()) {
```

Comment:
I think it's fine for now as is. My main concern beyond the churn was that there might be some disconnect over where we're headed with static vs. dynamic resources. I've created https://github.com/lyft/envoy-api/issues/112 to track.